js/index.js

```diff
@@ -36,14 +36,76 @@ window.addEventListener('load', event => {
     html += `</div>`;
   });
 
+console.log(memoryGame)
   // Add all the divs to the HTML
   document.querySelector('#memory-board').innerHTML = html;
 
   // Bind the click event of each element to a function
+  
+  let countClicks = 0
+
+  function flipCard (event) {
+
+    const board = event.target
+    const select = board.parentNode
+
+    if(select.className === 'card') {
+    select.className = 'card turned'
+    memoryGame.pickedCards.push(select.dataset.cardName)
+    console.log(memoryGame.pickedCards)
+    }
+      
+      if(memoryGame.pickedCards.length === 2) {
+    
+        let card1 = memoryGame.pickedCards[0]
+        let card2 = memoryGame.pickedCards[1]
+        console.log(card1)
+        console.log(card2)
+        if (memoryGame.checkIfPair(card1, card2)) {
+            document.querySelector('card turned div').className = 'blocked'
+          }
+          else {
+            document.querySelector('card turned div').className = 'card'
+          }
+     }  
+  }
+  
+  // for (i = 0; i < memoryGame.pickedCards.length; i++) {
+  //   checkIfPair(memoryGame.pickedCards[i])
+  //     if(true) {
+
+  //     } else {
+  //       turnCardsBack()
+  //     }
+  // }
+
+ function turnCardBack() {
+    select.className = 'card'
+  }
+
+  function printPairsClicked() {
+    countClicks ++
+    
+  }
+
+  function printPairsGuessed() {
+    if(card1.name === card2.name) {
+      this.pairsGuessed ++
+      let newPairsGuessedValue = this.pairsGuessed +1
```

Comment:
Solo vamos a usar "this" en las clases. Su uso fuera de una clase es más residual.